### DISCUSSION TITLE
How to Mask the connection Passwords in airflow logs

### DISCUSSION THREAD

**Discussion by Poovarasan-dv** [apache/airflow#24435] [Q&A]
As currently I am using the airflow version 2.2.4  config file in AKS cluster , I generate the fernet key and put into our config and ymal file.and I changed the **hide_sensitive_var_conn_fields = True** in our config file but still I can see the connections passwords in our config file

**Comment by potiuk** [comment]
Are you talking about config or log ? your title and message are contradictory. Can you please specifically explain what sensitive value you see where ? Which logs? Examples (redacted of cours) are great way to avoid asking readers to spend a lot of time on guessworking.

**Reply by Poovarasan-dv** [reply]
@potiuk I am asking about Mask passwords and sensitive info in task logs and UI #15599  I am using the  airflow version 2.2.4 with base config file So my requirement is my connection passwords are not showing in web UI logs  but now I can able to see the connections passwords in task logs i want to hide that. can you Please help this.

**Reply by potiuk** [reply]
Your question  is very chotic. 

You switch between config/web/ui/task logs and you say it works somewher but it does not work somewhere else. If you ned help, you need to explain clearly and precisely what does not work, and what works. It's you who want help. Spend your time on explaining it in the way that others can understand. Examples of logs and screenshots of what works and what does not work, please.

**Reply by Poovarasan-dv** [reply]
@potiuk  So what I am saying is I need to Mask the password in airflow task logs in web UI ,

- Using connection to: id: abc. Host: .azure, Port: 443, Schema: PROD, Login: AIRFLOW, **Password: abcd**, **extra: {'account': 'abc', 'warehouse': 'abc', 'database': 'PROD', 'schema': 'STG', 'role': 'abc'}**[0m

I want to hide this password while when I access in web UI, instead of abc I need ****

**Reply by potiuk** [reply]
> Using connection to: id: abc. Host: .azure, Port: 443, Schema: PROD, Login: AIRFLOW, Password: abcd, **extra: {'account': 'abc', 'warehouse': 'abc', 'database': 'PROD', 'schema': 'STG', 'role': 'abc'}**�[0m

Aha. Now you are very specific. This has been fixed in Airlfow 2.3.0 please upgrade/

https://github.com/apache/airflow/pull/22627

It would have saved a lot of time and back-forth if you mentioned that in your first message.

**Comment by Poovarasan-dv** [comment]
@potiuk Mask secret values from connections and variables

This masks secret values in logs for Connections and Variables.

It behaves as follows:

Connection passwords are always masked, where-ever they appear.

This means, if a connection has a password of a, then every a in
log messages would get replaced with ***

"Sensitive" keys from extra_dejson are also masked. Sensitive is
defined by the "existing" mechanism that the UI used, based upon the
name of the key.

"Sensitive" Variables are also masked.

but now The issues is password is not masked in log messages.

**Reply by potiuk** [reply]
> but now The issues is password is not masked in log messages.

WHICH LOG MESSAGES?

**Reply by potiuk** [reply]
It's the third time I am asking - which log messages you are talking about and asking your for examples.

**Reply by Poovarasan-dv** [reply]
In  dag task logs

**Comment by Poovarasan-dv** [comment]
@potiuk  Airflow-fernet-key-does-not-mask-credentials

I am using Apache Airflow 2.2.4 with Python 3.7 and run everything in Kubernetes Pods . When I add connections to airflow I do it via the GUI because this way the passwords were supposed to be encrypted. In order for the encryption to work I installed the python package "apache-airflow[crypto]" on my local machine and generated a Fernet Key that I then put in my airflow components. yaml's as the variable "AIRFLOW__CORE__FERNET_KEY: 'MY_KEY'". I also added the package "apache-airflow[crypto]" to my airflow repositories requirements.txt so that airflow can handle fernet keys.

**Reply by potiuk** [reply]
Just to repeat. Once you started to be very specific I can help you.  This has been fixed in Airlfow 2.3.0 please upgrade/

https://github.com/apache/airflow/pull/22627

It would have saved a lot of time and back-forth if you mentioned detals of what data in which logs is missing  in your first message. I've asked precisely for this information since the very beginning.
